### PR TITLE
Fix possible NULL pointer dereference.

### PR DESCRIPTION
**Pull request by vitek**
This caused SEGFAULT when moving sql3 db from 1.7 to 1.9. With this fix I've managed to use old db successfully.

### CODE REVIEW

**Review by williamcroberts — CHANGES_REQUESTED** [review]

File: src/lib/db.c
Comment (vitek) [reply src/lib/db.c:2172]: do you mean something like this?

```c
if (a) {
    CK_BYTE type = type_from_ptr(a->pValue, a->ulValueLen);
    if (type != TYPE_BYTE_INT_SEQ) {
        rv = _db_update_tobject_attrs(updb, tobj->id, tobj->attrs);
    }
} else {
   tobject_free(tobj);
   continue;
}
```

or let it fail?

Comment (williamcroberts) [reply src/lib/db.c:2172]: No need for an else clause, as we need it to perform the `tobject_free()` or a memory leak will ensue. We also need to step the database so we can get the next row item to check.

Comment (vitek) [reply src/lib/db.c:2172]: so we shouldn't check `rv` in this case, right?

```
        if (a && rv != CKR_OK) {
            goto error;
        }
```

Comment (vitek) [reply src/lib/db.c]: thanks, fixed

**Review by williamcroberts — APPROVED** [review]

**Review by williamcroberts — CHANGES_REQUESTED** [review]
So overall the diff is fine, but you need to squash your two commits into one and improve the commit message and add Fixes: #845 and don't forget to signoff on your commit. Thanks!

**Comment by vitek** [comment]
> So overall the diff is fine, but you need to squash your two commits into one and improve the commit message and add Fixes: #845 and don't forget to signoff on your commit. Thanks!



> So overall the diff is fine, but you need to squash your two commits into one and improve the commit message and add Fixes: #845 and don't forget to signoff on your commit. Thanks!

done

**Comment by sergiorussia** [comment]
@williamcroberts could u take a look? the fix is critical

**Comment by sergiorussia** [comment]
also tagging some other active maintainers (https://github.com/orgs/tpm2-software/people) cuz last William's activity was on August 11.

@AndreasFuchsTPM
@pdxjohnny
@JuergenReppSIT
@wxleong

guys could you take a look?
the bug prevents eg from upgrading to Ubuntu 24.04 LTS

ps: after merging this one we could close #849 as a duplicate

**Comment by williamcroberts** [comment]
I'm gonna start the release process this week.

**Pull request merged**
